addons/website_event_track/controllers/event_track.py

```diff
@@ -392,31 +392,59 @@ def event_track_proposal(self, event, **post):
-                tags.append(tag.id)
+        if not event.can_access_from_current_website():
+            raise NotFound()
+
+        #  Tags Handler
+        tagEntries = post['tags'].split(',')
+        tags_registered = []
+        for tag in tagEntries:
+            if tag:
+                tags_registered.append(int(tag))
+
+        #  On the form, contact_name and at least one of contact_phone and contact_email must be filled.
+        #  Therefore, empty contacts are not considered here. No contact is created if the user do not tick
+        #  the additional information checkbox on the track proposal form.
+        contact = False
+        if 'add_contact_information' in post:
+            contact = request.env['res.partner'].sudo().create({
+                'name': post['contact_name'],
+                'email': post['contact_email'],
+                'phone': post['contact_phone'],
+            })
 
         track = request.env['event.track'].sudo().create({
             'name': post['track_name'],
             'partner_name': post['partner_name'],
-            'partner_email': post['email_from'],
-            'partner_phone': post['phone'],
-            'partner_biography': plaintext2html(post['biography']),
+            'partner_email': post['partner_email'],
+            'partner_phone': post['partner_phone'],
+            'partner_biography': plaintext2html(post['partner_biography']),
+            'partner_company_name': post['partner_company_name'],
+            'partner_function': post['partner_function'],
             'event_id': event.id,
-            'tag_ids': [(6, 0, tags)],
+            'tag_ids': [(6, 0, tags_registered)],
             'user_id': False,
             'description': plaintext2html(post['description']),
-            'image': base64.b64encode(post['image'].read()) if post.get('image') else False
+            'image': base64.b64encode(post['image'].read()) if post.get('image') else False,
+            'partner_id': contact.id if contact else False,
+            'contact_phone': contact.phone if contact else False,
+            'contact_email': contact.email if contact else False,
         })
-        if request.env.user != request.website.user_id:
-            track.sudo().message_subscribe(partner_ids=request.env.user.partner_id.ids)
-        else:
-            partner = request.env['res.partner'].sudo().search([('email', '=', post['email_from'])])
-            if partner:
-                track.sudo().message_subscribe(partner_ids=partner.ids)
-        return request.render("website_event_track.event_track_proposal", {'track': track, 'event': event})
 
+        if request.env.user == request.website.user_id:
+            #  Public user is removed from subscription if he creates the track.
+            track.sudo().message_unsubscribe(partner_ids=request.env.user.partner_id.ids)
+
+        if contact and contact.email:
+            track.sudo().message_subscribe(partner_ids=[contact.id])
```

Comment:
So apparently in version 14.0 the public user was subscribed to the track. I thought this was undesired behavior but if it is common I can just remove the lines and only subscribe the contact if there is one?

Reply:
No I mean this is a good idea, but you may use mail_create_nosubscribe context key to avoid subscribing author to the record. Then use message_subscribe if not public user.

Reply:
Done